ml-agents/mlagents/trainers/trainer.py

```diff
@@ -5,13 +5,15 @@
 from mlagents.tf_utils import tf
 
 from collections import deque
+from queue import Queue
```

Comment:
Why do we need `queue.Queue` here instead of something lighter-weight like `collections.deque`? I don't think we need the synchronization that Queue provides (https://docs.python.org/3/library/queue.html)

Reply:
Not yet :) The intention was to enable the environment stepping and trainer advancing to run in separate threads (or even processes) for algorithms like APE-X. Eventually we'd want the trainer to wait until a Trajectory is available before doing any computation. I went with Queue since the method calls between Queue (and multiprocessing.Queue) are different from `deque`. 

We could also just use `deque` until the time comes to change over.

Reply:
Sounds good.

Reply:
It appears that `deque` is actually thread-safe as long as you're using just `get` and `put`, so I'll switch to that. Even if we go multi-thread we can still use `deque`, and it's faster. 

https://stackoverflow.com/questions/717148/queue-queue-vs-collections-deque

Reply:
Should anything still be using queue.Queue()? If not, you should remove the imports and see what's breaking.

Reply:
I think some of the tests still do - will switch them out